flatfs/flatfs.go

```diff
@@ -34,11 +34,12 @@ type Datastore struct {
 	path string
 	// length of the dir splay prefix, in bytes of hex digits
 	hexPrefixLen int
+	sync         bool
 }
 
 var _ datastore.Datastore = (*Datastore)(nil)
 
-func New(path string, prefixLen int) (*Datastore, error) {
+func New(path string, prefixLen int, sync bool) (*Datastore, error) {
```

Comment:
perhaps we should keep `New` as is, setting sync to `true` and just add another constructor:

```
NewSync(...) 
```
with the bool flag option. this prevents changing the interface on whoever depends on this code